test/base/src/androidTest/kotlin/io/realm/test/shared/MutableRealmTests.kt

```diff
@@ -147,4 +149,19 @@ class MutableRealmTests {
             }
         }
     }
+
+    @Test
+    fun findLatest_inLongHistory() {
+        runBlocking {
+            val child = realm.write { copyToRealm(Child()) }
+            for (i in 1..10) {
+                realm.write {
+                    findLatest(child)?.apply {
```

Comment:
Maybe `assertNotNull(findLatest(child))` would be cleaner? Updating the value doesn't really serve a purpose?

Reply:
Updating the object serves the purpose that we keep introducing new versions and show that we still find the new ones from the original one across all the updates. Further, it is the updates that triggers internal cleanup and closing of unreferenced versions until we implement #332. 

But I guess you have a point in asserting the version actually being there and not just silently ignore that situation 🐒

Reply:
I believe we also advance the version even if there are no changes? (But I could be wrong 🙈 )

Reply:
I guess you are right: https://github.com/realm/realm-kotlin/blob/master/test/base/src/androidTest/kotlin/io/realm/test/shared/RealmTests.kt#L88